pkg/webhook/authorizer/withselectors.go

```diff
@@ -0,0 +1,123 @@
+}
+
+// NewWithSelectorsChecker creates a new WithSelectorsChecker.
+func NewWithSelectorsChecker(ctx context.Context, log logr.Logger, clientSet kubernetes.Interface, clock clock.Clock) WithSelectorsChecker {
+	return &withSelectorsChecker{
+		ctx:       ctx,
+		log:       log.WithName("authorize-with-selectors-checker"),
+		clientSet: clientSet,
+		clock:     clock,
+
+		isPossible:    false,
+		nextCheckTime: ptr.To(clock.Now()),
+	}
+}
+
+type withSelectorsChecker struct {
+	ctx       context.Context
+	log       logr.Logger
+	clientSet kubernetes.Interface
+	clock     clock.Clock
+
+	isPossible    bool
+	nextCheckTime *time.Time
+}
+
+func (w *withSelectorsChecker) IsPossible() (bool, error) {
+	if w.nextCheckTime != nil && w.clock.Now().UTC().After(w.nextCheckTime.UTC()) {
+		enabled, mustCheckAgain, err := w.isAuthorizeWithSelectorsFeatureEnabled()
+		if err != nil {
+			return false, fmt.Errorf("failed checking whether 'AuthorizeWithSelectors' feature is enabled: %w", err)
+		}
+
+		w.isPossible = enabled
+		if mustCheckAgain {
+			w.nextCheckTime = ptr.To(w.clock.Now().UTC().Add(10 * time.Minute))
+			w.log.Info("Must check again, caching result", "nextCheckTime", w.nextCheckTime.String())
+		} else {
+			w.nextCheckTime = nil
+			w.log.Info("No need not check again")
+		}
+	}
+
+	return w.isPossible, nil
+}
+
+func (w *withSelectorsChecker) isAuthorizeWithSelectorsFeatureEnabled() (enabled bool, mustCheckAgain bool, err error) {
+	version, err := semver.NewVersion(w.clientSet.Version())
+	if err != nil {
+		return false, true, fmt.Errorf("failed parsing Kubernetes version %q to semver: %w", w.clientSet.Version(), err)
+	}
+
+	switch {
+	case versionutils.ConstraintK8sLess131.Check(version):
+		w.log.Info("Kubernetes version is lower than 1.31 -> feature gate did not exist", "authorizationWithSelectorsPossible", false)
+		return false, false, nil
+
+	case versionutils.ConstraintK8sGreaterEqual134.Check(version):
+		w.log.Info("Kubernetes version is at least 1.34 -> feature gate is GA and locked to 'enabled'", "authorizationWithSelectorsPossible", true)
+		return true, false, nil
+
+	default:
+		// Feature gate exists but is alpha/beta and may be disabled - we have to check whether it is enabled by making
+		// a dry-run call with a label selector field. If it is still present in the output, then the
+		// 'AuthorizeWithSelectors' feature is enabled. If it was disabled, kube-apiserver would have removed it in its
+		// response.
+		w.log.Info("Kubernetes version is between 1.31 and 1.33 -> feature gate is alpha/beta and may be disabled, must check")
+
+		sar := &authorizationv1.SubjectAccessReview{
+			Spec: authorizationv1.SubjectAccessReviewSpec{
+				User: "gardener-admission-controller",
+				ResourceAttributes: &authorizationv1.ResourceAttributes{
+					Namespace:     "default",
+					Verb:          "get",
+					Resource:      "shoots",
+					LabelSelector: &authorizationv1.LabelSelectorAttributes{RawSelector: "foo=bar"},
```

Comment:
Maybe not the best suggestion, but `foo=bar` would be a bit strange in productive code or logs. 

```suggestion
					LabelSelector: &authorizationv1.LabelSelectorAttributes{RawSelector: "feature-gates.gardener.cloud/AuthorizeWithSelectors=enabled"},
```